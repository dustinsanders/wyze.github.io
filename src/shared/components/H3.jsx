// @flow

import type { Style, VNode } from '../../types'
import { css } from 'glamor'
import preact from 'preact'

type Props = {
  children?: VNode,
  styles?: Style,
}

const h3: Style = css({
  fontSize: '1.15em',
})

const H3 = ({ children, styles }: Props): VNode => (
  <h3 {...css(h3, styles)}>{children}</h3>
)

H3.defaultProps = {
  children: null,
  styles: {},
}

export default H3
