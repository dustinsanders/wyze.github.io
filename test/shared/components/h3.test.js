// @flow

import H3 from '../../../src/shared/components/H3'
import preact from 'preact'
import snapshot from '../../helpers/snapshot'

const children = (
  // eslint-disable-next-line react/jsx-filename-extension
  <span>Hello</span>
)

const styles = {
  color: 'rebeccapurple',
}

describe('<H3 />', () => {
  it('renders', snapshot.bind(null, H3, { children }))
  it('renders styles', snapshot.bind(null, H3, { children, styles }))
})
