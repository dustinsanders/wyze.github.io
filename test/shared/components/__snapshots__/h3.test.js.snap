// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H3 /> renders 1`] = `
"<h3 data-css-1wg584y>
  <span>Hello</span>
</h3>"
`;

exports[`<H3 /> renders styles 1`] = `
"<h3 data-css-1qdbaiw>
  <span>Hello</span>
</h3>"
`;
